src/Hpack/Module.hs

```diff
@@ -46,7 +46,7 @@ getModules dir src_ = sortModules <$> do
   if exists
     then do
       src <- Directory.canonicalizePath (dir </> src_)
-      removeSetup src . map toModule <$> getModuleFilesRecursive src
+      removeSetup src . nub . map toModule <$> getModuleFilesRecursive src
```

Comment:
Should this use [`nubOrd`](https://www.stackage.org/haddock/lts-16.20/containers-0.6.2.1/Data-Containers-ListUtils.html#v:nubOrd) instead?

Reply:
I was considering this as well.  I'm not eager to require `containers >= 0.6` though.  We can add our own inline definition.

Reply:
I shouldn't have used `nub`, I know, it is evil.
Instead, since `getModules` returns a sorted list, one could have applied `uniq` in the end which just removes consecutive duplicates.
```haskell
uniq = concat . map head . group
```
Or even better, modify `sortModules` to remove duplicates.
```haskell
sortModules = map Module . Set.toAscList . Set.fromList . map unModule
```
If you dare to derive `Ord` for `Module`, then just:
```haskell
sortModules = Set.toAscList . Set.fromList 
```